docs/frontend-api/authentication.md

```diff
@@ -0,0 +1,112 @@
+But if you want some users to be able to access those restricted parts, this requires implementing some form of user authentication to your app, so that your users can create an account, and later log back into that account, to access your protected content.
+The implementing some form of user authentication in the shopsys framework use access tokens.
+
+Access Tokens are used in token-based authentication to allow an application to access an API. For example, a customer orders API method can read orders for only one (authorized) customer.
+Once an application has received an Access Token, it will include that token as a credential when making API requests. To do so, it should transmit the Access Token to the API as a Bearer credential in an HTTP Authorization header.
+
+[JWT](https://jwt.io/introduction/) tokens are used as access tokens in the shopsys framework.
+
+You can read more about access tokens in [this](https://auth0.com/docs/tokens/concepts/access-tokens) article.
+
+The access token has a short lifetime (15 minutes in the base).
+So that users do not have to log in again after such a short time, the Shopsys framework introduces implementation of refresh tokens that have a longer lifetime (14 days in the base).
+You can read more about refresh tokens [here](https://auth0.com/docs/tokens/concepts/refresh-tokens).
+
+## How to get a pairs of tokens
+
+To get a pair of tokens, you can use the mutation query Login with the input parameters email and password. 
+The query will return a pair of tokens, one access token and a second refresh token.
+
+Login query look like this:
+
+```json
+{ "query": "mutation {Login(input: {email: \"--EMAIL--\", password: \"--PASSWD--\"}) {accessToken, refreshToken}}"}
+```
+
+!!!
+You should always send a queries over a secure HTTPS protocol.
+
+Response look like this:
+
+```json
+{
+  "data":{
+    "Login":{
+      "accessToken":"X.X.X",
+      "refreshToken":"X.X.X"
+    }
+  }
+}
+```
+
+The access token carries basic user information such as its uuid, name, email and role list.
+
+If you're running an application in [docker](../installation/installation-using-docker-application-setup.md) and you have [allowed frontend api on first domain](./introduction-to-frontend-api.md), curl command to get pair of tokens may look like this:
+
+```sh
+  curl 
+    -X POST 
+    -H "Content-Type: application/json" 
+    -d '{"query": "mutation {Login(input: {email: \"--EMAIL--\", password: \"--PASSWD--\"}) {accessToken, refreshToken}}"}'
+    http://127.0.0.1:8000/graphql/
+```
+
+
+## How to access to restricted part
+
+If you need to access to the private section, you must send a Bearer credential in the Authorization header.
+
+The Authorization header look like this:
+
+```sh
+  curl
+    -H "Authorization: Bearer --ACCESS TOKEN--" 
+```
+
+The frontend API is set so that receiving the Autohrization header in the request, the user is authenticated by the `TokenAuthenticator` class.
```

Comment:
```suggestion
The frontend API is set so that receiving the Authorization header in the request, the user is authenticated by the `TokenAuthenticator` class.
```